tests/publiccloud/cloud_netconfig.pm

```diff
@@ -0,0 +1,48 @@
+# Copyright 2023 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+
+# Package: cloud-netconfig-{azure, gc2, gce}
+# Summary: This test ensures the consistency of cloud-netconfig's
+# functionality. The test shall be conducted on a VM in the cloud
+# infrastructure of a supported CSP.
+#
+# This test only contains a minimal functionality that needs to
+# be extended in the future.
+#
+# Maintainer: qa-c team <qa-c@suse.de>
+
+use Mojo::Base 'publiccloud::basetest';
+use Test::Assert 'assert_equals';
+use testapi;
+
+sub run {
+    my ( $self, $args ) = @_;
+    my $instance = $args->{my_instance};
+
+    my $cloud_netconfig_svc_status = $instance->ssh_script_run(
+        cmd => 'systemctl status cloud-netconfig.service' );
+    my $cloud_netconf_status_ok = $cloud_netconfig_svc_status == 0
+      || $cloud_netconfig_svc_status == 3 ? 1 : undef;
+    assert_equals( $cloud_netconf_status_ok, 1,
+        'cloud-netconfig.service status appears to be incorrect!' );
+    $instance->ssh_assert_script_run(
```

Comment:
I think that you could use directly the `cloud_netconfig_svc_status ` in the `ssh_assert_script_run` command, so removing lines 24-27.

Reply:
I had this before but it was not liked to do comparisons in Bash so I read it and do the comparison in Perl.
I need to compare it in any language, because the status of the service is usually 3, since it is triggered by the timer.

Reply:
Suggestion:
`systemctl status cloud-netconfig.service; [ $? -eq 0 -o $? -eq 3] && true || false`

Reply:
Thanks for the comment. I really appreciate your opinion.
I had something like this before, because it was shorter. But I was adviced to do it differently. If computational checks are not done on the SUT, but just data is being collected instead, it might also be a benefit on the other side.

From my perspective it doesn't matter much where it the check is done. If more reviewers like it to be changed I can change it, bot for the moment it appears like a nuance to me which doesn't seem to be worth doing another VR.

Reply:
As you prefer.

Reply:
It's preferred to do this in Perl.